### PR TITLE
Generic XML output, with all parameters #16

### PR DESCRIPTION
**Pull request by Freundschaft**

### CODE REVIEW

File: src/main/java/org/wikivoyage/listings/output/Xml.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        new XMLSimpleNode(node, "tag").attrib("k", name).attrib("v", value);
+    }
+
+    public String getDefaultExtension()
+    {
+        if (userDefined) {
+            return ".user-defined.xml";
+        } else {
+            return ".xml";
```
Comment (baturin) [inline src/main/java/org/wikivoyage/listings/output/Xml.java]: There will be a conflict with OsmAnd XML - files will have the same name. 
I propose to return ".generic.xml" here.

File: src/main/java/org/wikivoyage/listings/output/Xml.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import org.wikivoyage.listings.entity.WikivoyagePOI;
+import org.wikivoyage.listings.utils.XMLSimpleNode;
+import org.wikivoyage.listings.utils.XMLSimpleNodeException;
+
+public class Xml extends NavigationXMLOutputFormat {
+    private static Integer nodeId = 0;
+    private boolean userDefined;
+
+    public Xml(boolean userDefined) {
+        this.userDefined = userDefined;
```
Comment (baturin) [inline src/main/java/org/wikivoyage/listings/output/Xml.java]: user defined flag is not needed here, propose to remove it

File: src/main/java/org/wikivoyage/listings/output/Xml.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    .attrib("accessibility", poi.getAccessibility())
+                    .attrib("lastEdit", poi.getLastEdit())
+                    .attrib("description", poi.getDescription());
+            nodeId++;
+
+            if (!userDefined) {
+                addTagNode(poiNode, "wikivoyage", poi.getType());
+            } else {
+                addTagNode(poiNode, "user_defined", "user_defined");
+            }
```
Comment (nicolas-raoul) [inline src/main/java/org/wikivoyage/listings/output/Xml.java]: Not neede either.

File: src/main/java/org/wikivoyage/listings/output/Xml.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    public XMLSimpleNode createXml(WikivoyagePOI[] pois) throws XMLSimpleNodeException
+    {
+        XMLSimpleNode genericNode = new XMLSimpleNode("generic")
+                .attrib("version", "0.5")
+                .attrib("generator", "wikivoyage-pois-converter");
+
+        for (WikivoyagePOI poi: pois) {
+            XMLSimpleNode poiNode = new XMLSimpleNode(genericNode, "node")
+                    .attrib("id", nodeId.toString())
+                    .attrib("visible", "true")
```
Comment (nicolas-raoul) [inline src/main/java/org/wikivoyage/listings/output/Xml.java]: Where does this come from?
All of the rest looks good :-)

Comment (patrickwieth) [inline src/main/java/org/wikivoyage/listings/output/Xml.java]: hmm, this was in OsmXml.java, the "visible" attribute is only for Osm? If so, I can strip this also off

Comment (nicolas-raoul) [inline src/main/java/org/wikivoyage/listings/output/Xml.java]: Yes, please remove it and then it should be good to go.

File: src/main/java/org/wikivoyage/listings/output/Xml.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package org.wikivoyage.listings.output;
+
+import org.wikivoyage.listings.entity.WikivoyagePOI;
+import org.wikivoyage.listings.utils.XMLSimpleNode;
+import org.wikivoyage.listings.utils.XMLSimpleNodeException;
+
+public class Xml extends NavigationXMLOutputFormat {
+    private static Integer nodeId = 0;
+    private boolean userDefined;
```
Comment (nicolas-raoul) [inline src/main/java/org/wikivoyage/listings/output/Xml.java]: oops just noticed this, please remove thanks!

**Comment by nicolas-raoul** [comment]
Alexey, I hope you don't mind if I merge?

**Pull request merged**